lcdb/test/reporting_test.py

```diff
@@ -0,0 +1,206 @@
+    complectitur mea ex.
+
+    {Table1}
+
+    ### Sub section 2.2
+
+    Bulleted list
+
+    * one
+    * two
+    * thee
+
+    Sed sumo posse liberavisse ex. Eu ponderum neglegentur cum, ius eu tractatos
+    partiendo. Albucius explicari vim et. Id maluisset argumentum vim. Ex mei summo
+    conceptam, ut adipisci dignissim disputando per. At detracto intellegebat vim,
+    mea nihil everti at. Wisi error vis no, pri choro ceteros ne.
+
+    ### Sub section 2.3
+
+    Exerci laudem option pri te, vix modus inani signiferumque an, magna
+    conclusionemque has ex. Vitae graeco quo ex, labore qualisque vel in, at mei
+    eius nihil postea. Eum iudico audire eligendi ea, at eam malis aliquip. Vix at
+    feugiat postulant ullamcorper, ius no prodesset repudiandae. Hendrerit
+    complectitur mea ex.""")
+
+test_table = pd.DataFrame({'sampleID': ['treated1', 'treated2'], 
+                           'treatment': ['trt1', 'trt2'], 
+                           'replicate': ['1', '2']
+                           })[['sampleID', 'treatment', 'replicate']]
+
+def renderHtml(string):
+        from lcdb.reporting import ENV
+        with open('tmp_render.html', 'w') as OUT:
+            tp = ENV.get_template('base.html')
+            OUT.write(tp.render(content=string))
+            
+
+class TestReportImage(unittest.TestCase):
+    def setUp(self):
+        self.imageFile, _ = urlretrieve(test_imgURL)
+
+    def tearDown(self):
+        os.unlink(self.imageFile)
+
+    def test_Image(self):
+        from lcdb.reporting import Image
+        img = Image(self.imageFile)
+        self.assertEqual(str(img), test_imgB64)
+
+
+class TestReportPanel(unittest.TestCase):
+    def setUp(self):
+        self.imageFile, _ = urlretrieve(test_imgURL)
+
+    def tearDown(self):
+        os.unlink(self.imageFile)
+
+    def test_jinja_panel_with_caption(self):
+        from lcdb.reporting import JinjaPanel, Image, ENV
+        img1 = Image(self.imageFile, name='Sample1', caption='male')
+        img2 = Image(self.imageFile, name='Sample2', caption='female')
+        img3 = Image(self.imageFile, name='Sample3', caption='male')
+
+        table = JinjaPanel()
+        table.add_row([img1, img2, img3], caption='Row caption')
+        table.add_row([img1, img2, img3], caption='Row caption2')
+        self.assertEqual(table.panel, test_imagePanel)
+
+
+class Test_report(unittest.TestCase):
+    def setUp(self):
+        pass
+
+    def tearDown(self):
+        pass
+
+    def test_report(self):
+        from lcdb.reporting import report
+
+
+        values = {'Panel1': dedent(test_imagePanel),
+                  'Image1': dedent(test_imgB64_html),
+                  'Table1': test_table.to_html(index=False, classes='table table-striped')
+                }
+
+        with open('tmp_render.html', 'w') as OUT:
+            OUT.write(report(test_report, **values))
+
+if __name__ == '__main__':
+    unittest.main()
```

Comment:
The way I was generating the tests was using the eyeball test to get what I wanted then make the test string using that. Was taking too much time.